### PR TITLE
fix(Popup): Do not propagate keyboard event only when focus trap behavior is used

### PR DESCRIPTION
**Pull request by sophieH29**

### CODE REVIEW

File: packages/react/test/specs/components/Popup/Popup-test.tsx [hunk 1]
```diff
@@ -266,4 +263,44 @@ describe('Popup', () => {
...
+
+      // when popup open, check that stopPropagation is called when keyboard events are invoked
+      const stopPropagation = jest.fn()
+      const popupContentElement = getPopupContent(popup)
+      popupContentElement.simulate('keyDown', { stopPropagation })
+      if (shouldStopPropagation) {
+        expect(stopPropagation).toHaveBeenCalledTimes(1)
+      } else {
+        expect(stopPropagation).not.toBeCalled()
+      }
```
Comment (layershifter) [inline packages/react/test/specs/components/Popup/Popup-test.tsx]:
```
expect(stopPropagation).toHaveBeenCalledTimes(Number(shouldStopPropagation))
// or 
expect(stopPropagation).toHaveBeenCalledTimes(shouldStopPropagation ? 1 : 0)
```

File: packages/react/test/specs/components/Popup/Popup-test.tsx [hunk 2]
```diff
@@ -266,4 +263,44 @@ describe('Popup', () => {
...
+          content={<span id={contentId} />}
+          accessibility={behavior}
+        />,
+      )
+
+      // check popup open on key press
+      const popupTriggerElement = popup.find(`#${triggerId}`)
+      popupTriggerElement.simulate('keydown', { keyCode: keyboardKey.Enter })
+
+      expect(getPopupContent(popup).exists()).toBe(true)
```
Comment (layershifter) [inline packages/react/test/specs/components/Popup/Popup-test.tsx]: Not sure that we need this assertion there.

File: packages/react/test/specs/components/Popup/Popup-test.tsx [hunk 3]
```diff
@@ -266,4 +263,44 @@ describe('Popup', () => {
       expect(contentElement.classList.contains(Popup.Content.className)).toEqual(true)
     })
   })
+
+  describe('keyboard event propagation', () => {
+    const expectPopupToStopPropagation = (
```
Comment (mnajdova) [inline packages/react/test/specs/components/Popup/Popup-test.tsx]: Would rather to rename it to `expectPopupToHandleStopPropagation` as this tests both cases of when it should stop and not.

**Review by mnajdova — APPROVED** [review]
One small comment, other then that ❤️

File: packages/react/test/specs/components/Popup/Popup-test.tsx (hunk 1 above)
Comment (sophieH29) [reply packages/react/test/specs/components/Popup/Popup-test.tsx]: nice 👍

File: packages/react/test/specs/components/Popup/Popup-test.tsx (hunk 2 above)
Comment (sophieH29) [reply packages/react/test/specs/components/Popup/Popup-test.tsx]: you're right, will remove

File: packages/react/test/specs/components/Popup/Popup-test.tsx (hunk 3 above)
Comment (sophieH29) [reply packages/react/test/specs/components/Popup/Popup-test.tsx]: will rename

**Review by layershifter — APPROVED** [review]

**Pull request merged**